### PR TITLE
add explicit check to prevent member injection in kotlin objects

### PR DESCRIPTION
**Pull request by trevjonez**
resolves root cause of issue #1665

### CODE REVIEW

File: javatests/dagger/internal/codegen/BUILD [hunk 1]
```diff
@@ -37,6 +45,7 @@ GenJavaTests(
     plugins = ["//java/dagger/internal/codegen/bootstrap"],
     deps = [
         ":kotlin_injected_qualifier",
+        ":kotlin_object_with_member_injection",
```
Comment (trevjonez) [inline javatests/dagger/internal/codegen/BUILD]: should these be folded into a single lib? naming was very specific so I went this way, just not sure what is preferred.

File: java/dagger/internal/codegen/validation/InjectValidator.java [hunk 2]
```diff
@@ -332,6 +340,15 @@ private void checkInjectIntoPrivateClass(
     }
   }
 
+  private void checkInjectIntoKotlinObject(
+      TypeElement element, ValidationReport.Builder<TypeElement> builder) {
+    if (kotlinMetadataUtil.isObjectClass(element)) {
```
Comment (danysantiago) [inline java/dagger/internal/codegen/validation/InjectValidator.java]: We should also check if its a Companion Object:
`kotlinMetadataUtil.isObjectClass(element) || kotlinMetadataUtil.isCompanionObjectClass(element)`

File: javatests/dagger/internal/codegen/BUILD (hunk 1 above)
Comment (danysantiago) [reply javatests/dagger/internal/codegen/BUILD]: This is fine for now, if it gets too big we can consider folding.

File: java/dagger/internal/codegen/validation/InjectValidator.java [hunk 3]
```diff
@@ -332,6 +340,15 @@ private void checkInjectIntoPrivateClass(
...
+      builder.addError(
+          "Dagger does not support injection into kotlin objects",
```
Comment (danysantiago) [inline java/dagger/internal/codegen/validation/InjectValidator.java]: nit: Uppercase 'K', for Kotlin

**Comment by danysantiago** [comment]
Thank you so much Trevor, I'll try to get this submitted internally.

**Comment by trevjonez** [comment]
No big deal, thanks for the quick responses!

**Pull request closed without merging**